model/version.go

```diff
@@ -629,6 +632,43 @@ func GetVersionsWithOptions(projectName string, opts GetVersionsOptions) ([]Vers
 	return res, nil
 }
 
+// ModifyVersionsOptions is a struct containing options necessary to modify versions.
+type ModifyVersionsOptions struct {
+	Priority      *int64 `json:"priority"`
+	StartTimeStr  string `json:"start_time_str"`
+	EndTimeStr    string `json:"end_time_str"`
+	RevisionStart int    `json:"revision_start"`
+	RevisionEnd   int    `json:"revision_end"`
+	Requester     string `json:"requester"`
+}
+
+// GetVersionsToModify returns a slice of versions intended to be modified that satisfy the given ModifyVersionsOptions.
+func GetVersionsToModify(projectName string, opts ModifyVersionsOptions, startTime, endTime time.Time) ([]Version, error) {
+	projectId, err := GetIdForProject(projectName)
+	if err != nil {
+		return nil, err
+	}
+	match := bson.M{
+		VersionIdentifierKey: projectId,
+		VersionRequesterKey:  opts.Requester,
+	}
+
+	// setting revision numbers will take precedence over setting start and end times
+	if opts.RevisionStart > 0 {
+		match[VersionRevisionOrderNumberKey] = bson.M{"$lte": opts.RevisionStart, "$gte": opts.RevisionEnd}
```

Comment:
Do we want revisionEnd / endTime required or would we just allow this to default to latest?

Reply:
This is still a question I have^